doc/architecture/domains/transfer_agreement_automation.md

```diff
@@ -0,0 +1,54 @@
+## Transfer Agreement Automation flow
+
+First the deposit endpoint is exchanged between receiver and sender. The receiver creates an endpoint in his wallet. Then a Base64 (or what we choose) encoded string of the wallet deposit endpoint is handed to the sender, which then is created on the senders wallet. A reference guid to the receiver wallet is returned to the sender which is used by the transfer agreement automation background service.
+
+```mermaid
+sequenceDiagram
+    actor receiver as Receiver
+    actor sender as Sender
+    participant ta as Transfer Agreement API
+    box grey Same wallet system
+        participant senderWallet as Sender Wallet
+        participant wallet as receiver Wallet
+    end
+
+    receiver ->>+ ta: POST api/transfer-agreement/create-wallet-deposit-endpoint (OBS! should be moved to a wallet backend in the future)
+
+    ta ->>+ wallet: gRPC CreateWalletDepositEndpoint()
+    wallet -->>- ta: WalletDepositEndpoint
+
+    ta -->>- receiver: base64 WalletDepositEndpoint
+
+    receiver ->> sender: base64 WalletDepositEndpoint (over coffee)
+
+    sender ->>+ ta: POST api/transfer-agreement/create
```

Comment:
I dont think we should have a post endpoint call create. could we expand the existing one to take a TA or a base64 string?

Reply:
This is not ment to be the exact name for the call, but sure I can rename it